src/TableauConnector.js

```diff
@@ -83,12 +84,12 @@ class TableauConnector {
       const code = this.code
       return auth.getToken(code).then(accessToken => {
         // Restore canonical WDC URL, which Tableau saves with data source
-        const canonicalQueryString = Object.keys(this.params)
-          .filter(key => !!this.params[key])
-          .reduce((prev, key) => {
-            return `${prev ? prev + '&' : '?'}${encodeURIComponent(key)}=${encodeURIComponent(this.params[key])}`
-          }, '')
-        window.location = `/${canonicalQueryString}`
+        const parsedQueryString = queryString.parse(location.search)
+        let canonicalQueryString = `/?state=${parsedQueryString.state}`
```

Comment:
`state`, assuming it might be an ID for a local storage object, is not the best thing to keep in the canonical URL. Remember, the canonical URL should not contain anything that can be stale or linked to one specific machine. It needs to work well as a starting point for reconfiguring existing data sources, no matter where/when the workbook is opened. 

That said, I recognize at this point that the query is still stored in local storage and will need to be retrieved post redirect and for that reason `state` is required in the canonical URL.

Please make sure, that there are no unintended side-effect. Specifically:
1. If the user modifies the text of the query by hand and saves the connection. When the same user edits the connection using the same machine, that the query they edit is the one they've customized by hand and not the one that can still be found in local storage.
2. If the user simply saves the connection. When another user (on another machine) edits the connection, that the query they edit is the one they've saved even though the state is no longer found in local storage.